contracts/ERC721A.sol

```diff
@@ -381,11 +446,30 @@ contract ERC721A is IERC721A {
         // balance or numberMinted overflow if current value of either + quantity > 1.8e19 (2**64) - 1
         // updatedIndex overflows if _currentIndex + quantity > 1.2e77 (2**256) - 1
         unchecked {
-            _addressData[to].balance += uint64(quantity);
-            _addressData[to].numberMinted += uint64(quantity);
-
-            _ownerships[startTokenId].addr = to;
-            _ownerships[startTokenId].startTimestamp = uint64(block.timestamp);
+            // Updates:
+            // - `balance += quantity`.
+            // - `numberMinted += quantity`.
+            //
+            // We can directly add to the balance and number minted.
+            _packedAddressData[to] += quantity | (quantity << 64);
+
+            { // Scoped for extra gas optimization.
+                uint256 packed;
```

Comment:
Can we move this block of code into a helper function and reference it here and in _safeMint so we can reduce code duplication?

Reply:
Ok, fortunately, the compiler will inline them properly.